File: python3/july/day_19_Add Binary.py
class Solution:
    def addBinary(self, a: str, b: str) -> str:
        res = ""
        i = 0
        a = a[::-1]
        b = b[::-1]
        c = 0
        while(i < len(a) or i < len(b)):
            s = c
            if(i < len(a)):
                s += 1 if a[i] == '1' else 0

            if(i < len(b)):
                s += 1 if b[i] == '1' else 0

            c = s // 2
            s = s % 2
            res += '1' if s else '0'
            i += 1

        if(c):
            res += '1'

        return res[::-1]

File: cpp/july/day_19_Add Binary.cpp
class Solution {
public:
    string addBinary(string a, string b) {
        string res = "";
        int i = 0, j, c, s;
        reverse(a.begin(), a.end());
        reverse(b.begin(), b.end());
        c = 0;
        while(i < a.size() || i < b.size()){
            s = c;
            if(i < a.size()){
                s += a[i] == '1' ? 1 : 0;
            }
            if(i < b.size()){
                s += b[i] == '1' ? 1 : 0; 
            }
            c = s / 2;
            s = s % 2;
            res += s ? '1' : '0';
            i++;
        }
        if(c){
            res += '1';
        }
        reverse(res.begin(), res.end());
        return res;
    }
};
